tests/admin_views/tests.py

```diff
@@ -3752,6 +3860,63 @@ def test_simple_inline(self):
         self.assertEqual(Widget.objects.count(), 1)
         self.assertEqual(Widget.objects.all()[0].name, "Widget 1 Updated")
 
+    def test_simple_inline_permissions(self):
+        "A simple model doesn't perform changes without change permissions for related object"
+        result = self.client.login(username='permissionuser', password='secret')
+        self.assertEqual(result, True)
+
+        # Without add permission, we can't add a new inline
+        self.post_data['widget_set-0-name'] = "Widget 1"
+        collector_url = reverse('admin:admin_views_collector_change', args=(self.collector.pk,))
+        response = self.client.post(collector_url, self.post_data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Widget.objects.count(), 0)
+
+        # But after getting the permisson, we can
+        self.permissionuser.user_permissions.add(get_perm(Widget, get_permission_codename('add', Widget._meta)))
+
+        self.post_data['widget_set-0-name'] = "Widget 1"
+        collector_url = reverse('admin:admin_views_collector_change', args=(self.collector.pk,))
+        response = self.client.post(collector_url, self.post_data)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(Widget.objects.count(), 1)
+        self.assertEqual(Widget.objects.all()[0].name, "Widget 1")
+        widget_id = Widget.objects.all()[0].id
+
+        # Check that the PK link exists on the rendered form
+        response = self.client.get(collector_url)
+        self.assertContains(response, 'name="widget_set-0-id"')
+
+        # Now resave that inline
```

Comment:
Indeed this doesn't test much... I can't see why it's there neither. I'll remove it.

I think it's worth testing the 302 response, as it may be that the query is successful, but redirects to a wrong page (e.g. if you don't have view permission and were redirected to change page, you'd get a 403)